network/stream/peer.go

```diff
@@ -47,8 +47,8 @@ type Peer struct {
 	quit chan struct{} // closed when peer is going offline
 }
 
-// NewPeer is the constructor for Peer
-func NewPeer(peer *network.BzzPeer, baseKey []byte, i state.Store, providers map[string]StreamProvider) *Peer {
+// newPeer is the constructor for Peer
+func newPeer(peer *network.BzzPeer, address *network.BzzAddr, i state.Store, providers map[string]StreamProvider) *Peer {
```

Comment:
BzzPeer already has BzzAddr. Maybe it is possible to remove address argument?

Reply:
If I am not mistaken, peer has a BzzAddr for a peer, and address is a base address. I will rename the argument to `baseAddress` to avoid the confusion.